### PR TITLE
Py3 Skare3 and reorg

### PR DESCRIPTION
**Pull request by jeanconn**
This has 

1) 2to3 python changes (and few manual tweaks)
2) Tweaks for modern Perl in skare3
3) Moves "static" data (except task_schedule.cfg) and bin/ into share/arc3 
4) Puts data in arc3 (with the idea that this could run in ska3/flight with shared data link in parallel with ska2 version

### CODE REVIEW

**Comment by jeanconn** [comment]
This also requires updates to Ska::RDB and Ska::Web for modern Perl.  I've got those in my dev ska but will also make PRs for those bits too.

**Comment by jeanconn** [comment]
Seemed like it just made more sense to start to move forward to skare3 than to try to add more patches for broken SSL in ska2 Perl.

**Comment by taldcroft** [comment]
I thought I'd have a go at bringing this home, but it looks to be out of my league.

I installed a fresh ska3-flight and ska3-perl from $ska3conda/test (success!), but 

```
  573  source activate arc3
  574  export SKA=/export/tom/anaconda/envs/arc3
  579  source /export/tom/anaconda/envs/arc3/bin/ska_envs.sh 
  584  cd $SKA/share/arc3
  586  ./get_web_content.pl
  587  ./get_iFOT_events.pl 
```
```
(arc3) kadi$ echo $PERL5LIB
/export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2:/export/tom/anaconda/envs/arc3/lib/5.26.2:/export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/x86_64-linux

(arc3) kadi$ ./get_iFOT_events.pl 
Can't load '/export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/x86_64-linux/auto/HTML/Parser/Parser.so' for module HTML::Parser: /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/x86_64-linux/auto/HTML/Parser/Parser.so: undefined symbol: PL_curpad at /export/tom/anaconda/envs/arc3/lib/5.26.2/XSLoader.pm line 96.
 at /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/x86_64-linux/HTML/Parser.pm line 11.
Compilation failed in require at /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/HTML/TableExtract.pm line 17.
BEGIN failed--compilation aborted at /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/HTML/TableExtract.pm line 17.
Compilation failed in require at ./get_iFOT_events.pl line 7.
BEGIN failed--compilation aborted at ./get_iFOT_events.pl line 7.
```

```
(arc3) kadi$ ./get_web_content.pl
Error. You have specified 'Python' as an Inline programming language.

I currently only know about the following languages:
    C, Foo, foo

If you have installed a support module for this language, try deleting the
config-x86_64-linux-thread-multi-5.026002 file from the following Inline DIRECTORY, and run again:

    /export/tom/anaconda/envs/arc3/share/arc3/_Inline

(And if that works, please file a bug report.)

 at /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/Ska/Convert.pm line 13.
BEGIN failed--compilation aborted at /export/tom/anaconda/envs/arc3/lib/site_perl/5.26.2/Ska/Convert.pm line 32.
Compilation failed in require at ./get_web_content.pl line 9.
BEGIN failed--compilation aborted at ./get_web_content.pl line 9.
```

**Comment by jeanconn** [comment]
Gah.  Sorry, for the newer Ska::Convert that uses Inline::Python to call Chandra.Time we probably need to modify flt_envs to explicitly set PYTHONHOME to SKA.

The only "client" of the updated Ska::Convert thus far has been starcheck which is setting PYTHONHOME it its own wrapper, so I hadn't hit this yet.

Sorry the perl is still pretty Alpha.

**Comment by jeanconn** [comment]
Or actually, I will just need to research PYTHONHOME some more.  It may be that the better answer is to set it explicitly in a perl wrapper (like perlska3) or an application specific perl wrapper if it is contraindicated for our general ska3 environment.

**Comment by taldcroft** [comment]
Yes, generally setting PYTHONHOME is a non-starter.

**Comment by jeanconn** [comment]
Do you recall what it will screw up?  Will it be acceptable for all mixed applications to set it?

**Comment by jeanconn** [comment]
I also wonder if we can patch Inline::Python to work correctly without setting PYTHONHOME (but hard to know how much time to throw at Perl compatibility issues if in the end we will have just arc and starcheck using this Inline stuff).

**Comment by jeanconn** [comment]
With the updated Ska::Convert, this gets further.  That's with ska3-perl 2019.03.19 + perl-ska-web 4.0 .  I haven't immediately figured out:

```
(arc3) bash-4.1$ ./plot_hrc.py --h5=${SKA}/data/hrc_shield.h5 --out=${SKA}/www/ASPECT/arc3/hrc_shield.png
Segmentation fault (core dumped)
```

**Comment by jeanconn** [comment]
It looks to me like the segfault is one folks have seen before with zlib pytables matplotlib.  
```
https://github.com/ContinuumIO/anaconda-issues/issues/7315
```

File: arc3.cfg [hunk 1]
```diff
@@ -111,19 +111,15 @@ query_name = eclipse
...
+  scs131 = SCS 131       : SCS 131   :  Pitch Mom.
+  scs132 = SCS 132       : SCS 132   : Ephin
+  scs133 = SCS 133       : SCS 133   : EIO
   scs107 = SCS 107       : SCS 107   : EPH temp
   hrcshield = HRC shield : SHLD RT   : FSS Alfa
   hrcmcp = HRC MCP       : EVT RT    : AOACSTAT
-  E150   = E150          : E150      : SCS 130
-  E1300  = E1300         : E1300     : Pitch Mom
-  P4GM   = P4GM          : P4GM      : Yaw Mom
-  P41GM  = P41GM         : P41GM     : EPH temp
```
Comment (jeanconn) [inline arc3.cfg]: @taldcroft I've forgotten the history here; should values for these be fetched from a non-Snapshot if they are no longer in the snapshot?

Comment (jeanconn) [reply arc3.cfg]: I guess the P4GM and P41GM values are grabbed from GOES, but I'm not sure what the CXO values for E150 and E1300 should be at this point around

https://github.com/sot/arc/blob/9b314477240e03428b2fab6a5d342233f8391a57/arc.pl#L824

Comment (taldcroft) [reply arc3.cfg]: EPHIN is no more, off, kaput.  😢 :  So the E150 and E1300 table rows can be deleted entirely. :smile:

Comment (jeanconn) [reply arc3.cfg]: I think I'll leave the CXO values as '---' for now so until I understand the formatting issues later.

**Comment by taldcroft** [comment]
I'd forgotten about this, thanks for working on this migration!

File: arc3.cfg (hunk 1 above)
Comment (taldcroft) [reply arc3.cfg]: Yes, `---` is appropriate since there are no EPHIN (CXO) values but we do have meaningful proxies.

Comment (taldcroft) [reply arc3.cfg]: I meant for the P4 ones.

Comment (jeanconn) [reply arc3.cfg]: And yes, I figured.  And I'd originally meant to just use '---' for the E150 and E1300 rows, but I figured out where they live in the page template so have cut them successfully ineb1d586612afe1057d4212201d5e90fd806885b4

**Comment by jeanconn** [comment]
Thanks @taldcroft!  I just needed to work on something different for a few minutes.  For me, it looks like this is all actually working now that I've figured out that weird segfault when trying to make the hrc plots.  What are your thoughts on testing and promotion?  I'd certainly like to run "on the side" for a while before letting this take over, but it could still be safe to *run* it from /proj/sot/ska3/flight .

I could also conda or Python package it, but was not seeing a huge driver for that now (doesn't need to run on non HEAD).

**Comment by jeanconn** [comment]
Oh, and there's one defect... I haven't figured out what funny character has found its way into Snap.pm that is causing git to identify as binary.  If you review changes in detail please make sure to expand that diff and also please feel free to give input about whatever is wrong with the file.

**Comment by jeanconn** [comment]
I'm running this from /proj/sot/ska3/arc3test now.  Looks like the solar_wind.gif is still broken and the task_schedule should probably be in share instead of data.

**Comment by jeanconn** [comment]
Test outputs up at http://cxc.cfa.harvard.edu/mta/ASPECT/arc3/

File: get_goes_x.py [hunk 2]
```diff
@@ -18,17 +18,20 @@
 
 url = 'ftp://ftp.swpc.noaa.gov/pub/lists/xray/Gp_xr_5m.txt'
 
+last_err = None
```
Comment (jeanconn) [inline get_goes_x.py]: Is just adding a var outside the try/loop acceptable?  I'm not sure if this code worked in Py2, but in Py3 without some change if the tries fail I just get an error that "err" is undefined when the print statement tried to do its thing.

**Comment by jeanconn** [comment]
@taldcroft Do you have thoughts on getting LR approval to install this in ska3/flight today?  It lives within arc3 dirs .

**Comment by jeanconn** [comment]
I'm going to merge this and cut at version 4.0 and we can do any needed repairs with new PRs.

**Pull request merged**